aws-rotate-keys/orb.yml

```diff
@@ -43,4 +43,20 @@ commands:
             curl --fail https://circleci.com/api/v1.1/project/github/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/envvar?circle-token=<< parameters.circleci-token >> \
               -X POST --header "Content-Type: application/json" \
               -d "{ \"name\":\"<< parameters.aws-secret-access-key-var >>\", \"value\":\"${new_secret_access_key}\" }" &&
-            aws iam delete-access-key --user-name << parameters.aws-username >> --access-key-id ${old_access_key_id}
+            aws iam delete-access-key --user-name << parameters.aws-username >> --access-key-id ${old_access_key_id} &&
+            echo &&
+            # Although the new token is visible on the aws API, aws does not accept it
+            # for authentication immediately.
+            echo 'export AWS_ACCESS_KEY_ID=$new_access_key_id' >> $BASH_ENV &&
+            echo 'export AWS_SECRET_ACCESS_KEY=$new_secret_access_key' >> $BASH_ENV &&
+            source $BASH_ENV &&
+            echo Testing new token... &&
+            while true; do \
+              aws iam list-access-keys \
```

Comment:
If the new access key is not yet available wouldn't this command exit with an error code and break the loop (since `-eo pipefail` are the default shell options on CircleCI)?

Reply:
I think
- `-o pipefail` has no affect, since we don't do piping
- but I am not sure why `-e` doesn't catch it.
I tested it in CircleCI and it works. Would you like me to dig deeper?

Reply:
LGTM, but it would be good to understand why the `-e` option doesn't have any effect for failures.

Reply:
Here is why: https://unix.stackexchange.com/questions/312631/bash-script-with-set-e-doesnt-stop-on-command